### PR TITLE
v6.3: Add description about SM3 encryption authentication

### PR DESCRIPTION
**Pull request by CbcWestwolf**
### What is changed, added or deleted? (Required)

Add the description about `tidb_sm3_password` authentication and `SM3()` function.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v6.3 (TiDB 6.3 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/10578
- Other reference link(s): https://github.com/pingcap/tidb/pull/36193

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: basic-features.md [hunk 1]
```diff
@@ -134,17 +134,18 @@ This document lists the features supported in each TiDB version. Note that suppo
...
-| [Role-based authentication (RBAC)](/role-based-access-control.md) | Y | Y | Y | Y    |    Y    |    Y    |    Y    |    Y    |    Y    |
-| [Certificate-based authentication](/certificate-authentication.md) | Y | Y | Y | Y    |    Y    |    Y    |    Y    |    Y    |    Y    |
-| `caching_sha2_password` authentication                       | Y | Y | Y | Y    |    Y    |    Y    |    N    |    N    |    N    |
-| [MySQL compatible `GRANT` system](/privilege-management.md)  | Y | Y | Y | Y    |    Y    |    Y    |    Y    |    Y    |    Y    |
-| [Dynamic Privileges](/privilege-management.md#dynamic-privileges) | Y | Y | Y | Y    |    Y    |    Y    |    Y    |    N    |    N    |
-| [Security Enhanced Mode](/system-variables.md#tidb_enable_enhanced_security) | Y | Y | Y | Y    |    Y    |    Y    |    Y    |    N    |    N    |
-| [Redacted Log Files](/log-redaction.md)                      | Y | Y | Y | Y    |    Y    |    Y    |    Y    |    Y    |    N    |
+| Security                                               | 6.3 | 6.2 | 6.1 | 6.0 | 5.4  | 5.3 | 5.2 | 5.1 | 5.0 | 4.0 |
```
Comment (ran-huang) [inline basic-features.md:137]: Please align with https://github.com/pingcap/docs-cn/pull/10578.
Some commits are not included in this PR.

File: security-compatibility-with-mysql.md [hunk 2]
```diff
@@ -37,6 +37,7 @@ The support for TLS authentication is configured differently. For detailed infor
 | `sha256_password`        | No               |
 | `caching_sha2_password`  | Yes, since 5.2.0 |
 | `auth_socket`            | Yes, since 5.3.0 |
+| `tidb_sm3_password`      | Yes, since 6.3.0 |
 | [TLS Certificates]       | Yes              |
```
Comment (ran-huang) [inline security-compatibility-with-mysql.md:41]:
```suggestion
| [TLS Certificates](/enable-tls-between-clients-and-servers.md)       | Yes              |
```

File: system-variables.md [hunk 3]
```diff
@@ -206,7 +206,7 @@ mysql> SELECT * FROM t1;
 - Persists to cluster: Yes
 - Type: Enumeration
 - Default value: `mysql_native_password`
-- Possible values: `mysql_native_password` and `caching_sha2_password`
+- Possible values: `mysql_native_password`, `caching_sha2_password` and `tidb_sm3_password`
```
Comment (ran-huang) [inline system-variables.md:209]:
```suggestion
- Possible values: `mysql_native_password`, `caching_sha2_password`, and `tidb_sm3_password`
```

**Review by ran-huang — APPROVED** [review]
Rest LGTM

File: security-compatibility-with-mysql.md [hunk 4]
```diff
@@ -37,7 +37,8 @@ The support for TLS authentication is configured differently. For detailed infor
 | `sha256_password`        | No               |
 | `caching_sha2_password`  | Yes, since 5.2.0 |
 | `auth_socket`            | Yes, since 5.3.0 |
-| [TLS Certificates]       | Yes              |
+| `tidb_sm3_password`      | Yes, since 6.3.0 |
+| [TLS Certificates](/enable-tls-between-clients-and-servers.md)       | Yes              |
```
Comment (Oreoxmt) [inline security-compatibility-with-mysql.md:41]:
```suggestion
| TLS Certificates      | Yes              |
```

**Review by Oreoxmt — APPROVED** [review]
Rest LGTM

**Pull request merged**